Bottom: 344d2a308759f01616a248f648569c7167b24e41
Top:    178c7f657abdc27983f53804fd87eb3139029fa8
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-06 00:22:51 +0100

Add a HID API call, notify_pcb_filename_change()

This is so the core can tell the GUI when the PCB being edited is saved
into a different file.


---

diff --git a/src/action.c b/src/action.c
index 3756bc4..c375805 100644
--- a/src/action.c
+++ b/src/action.c
@@ -5684,6 +5684,8 @@ ActionSaveTo (int argc, char **argv, Coord x, Coord y)
     {
       if (SavePCB (name) == 0)
         {
+          if (gui->notify_pcb_filename_change != NULL)
+            gui->notify_pcb_filename_changed (PCB->Filename, name);
           free (PCB->Filename);
           PCB->Filename = strdup (name);
           SetChangedFlag (false);
diff --git a/src/command.c b/src/command.c
index f7a4983..4ed7ef5 100644
--- a/src/command.c
+++ b/src/command.c
@@ -351,6 +351,8 @@ CommandSaveLayout (int argc, char **argv, Coord x, Coord y)
     case 1:
       if (SavePCB (argv[0]) == 0)
         {
+           if (gui->notify_pcb_filename_change != NULL)
+            gui->notify_pcb_filename_changed (PCB->Filename, argv[0]);
           free (PCB->Filename);
           PCB->Filename = strdup (argv[0]);
           SetChangedFlag (false);
diff --git a/src/hid.h b/src/hid.h
index b1f532f..322602a 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -567,6 +567,10 @@ typedef enum
      * Any remaining rendering will be flushed to the screen.
      */
     void (*finish_debug_draw)  (void);
+
+    /* Notification to the GUI that the PCB file is being renamed. */
+    void (*notify_pcb_filename_change) (const char *old_filename,
+                                        const char *new_filename);
   };
 
 /* Call this as soon as possible from main().  No other HID calls are
